src/App.tsx

```diff
@@ -1,35 +1,178 @@
+    setCurrentTodos([
+      ...currentTodos,
+      {
+        id: getNewTodoId(currentTodos),
+        title,
+        completed: false,
+        userId,
+        user: getUserById(userId),
+      },
+    ]);
+
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
+    <div className="section">
+      <h1 className="title">Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        className="box"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label className="label" htmlFor="title">
+            Title:
+          </label>
+          <div
+            className={cn('control', {
+              'has-icons-right': hasTitleError,
+            })}
+          >
+            <input
+              id="title"
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              className={cn('input', {
+                'is-danger': hasTitleError,
+              })}
+              value={title}
+              onChange={handleTitleChange}
+            />
+
+            {hasTitleError && (
+              <span className="icon is-small is-right">
+                <i className="fas fa-exclamation-triangle has-text-danger"></i>
+              </span>
+            )}
+          </div>
+          {hasTitleError && (
+            <p className="help is-danger">Please enter a title</p>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
+          <label className="label" htmlFor="user">
+            User:
+          </label>
+          <div className="control has-icons-left">
+            <div
+              className={cn('select', {
+                'is-danger': hasUserIdError,
+              })}
+            >
+              <select
+                data-cy="userSelect"
+                id="user"
+                value={userId}
+                onChange={handleUserChange}
+              >
+                <option value="0" disabled>
+                  Choose a user
+                </option>
+                {usersFromServer.map((user: User) => (
+                  <option value={user.id} key={user.id}>
+                    {user.name}
+                  </option>
+                ))}
```

Comment:
Ensure that `usersFromServer` is correctly populated and handles potential errors or empty states, as this could affect the user selection dropdown.